### PR TITLE
feat: add `int` and `float` Avro type support

### LINKED ISSUE #103: Add types.Int32
It's nice to have explicit support fo Avro `int` type for compatibility with existing software

```
class Int32(int):
    ...
```

### PR DESCRIPTION
**Pull request by judahrand**
Closes #107
Closes #103

### CODE REVIEW

**Comment by judahrand** [comment]
@marcosschroh Thoughts? I think the `typing.Annotated` approach makes a lot more sense than the other PR as all we are aiming to use the native type but communicate addition information to the package to Avro serialization + schema generation.

**Comment by eugene-bright** [comment]
Looks much cooler than my approach, thank you!

BTW, you can automate closing related issues by [proper pull request description](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

**Comment by eugene-bright** [comment]
I think @marcosschroh will ask for documentation update

**Comment by judahrand** [comment]
> BTW, you can automate closing related issues by proper pull request description.

Huh, didn't actually know that! Thanks!

> I think @marcosschroh will ask for documentation update

There is a bit of docs in here. Wasn't really sure what else to add!

File: docs/fields_specification.md [hunk 1]
```diff
@@ -15,27 +15,32 @@ The set of primitive type names is:
...
-| bytes     |     bytes   |
+| Avro Type    | Python Type |
+|--------------|-------------|
+| string       |     str     |
+| int,long     |     int     |
+| boolean      |     bool    |
+| float,double |     float   |
+| null         |     None    |
+| bytes        |     bytes   |
+
```
Comment (marcosschroh) [inline docs/fields_specification.md:26]: Good explanation. Maybe we can also add two new rows for the new types in order to make it more clear for the users, what do you think?  

| Avro Type | Python Type |
|-----------|-------------|
| string    |     str     |
| long      |     int     |
| boolean   |     bool    |
| double    |     float   |
| null      |     None    |
| bytes     |     bytes   |
|  float  |    types.Float32   |
|   int   |   types.Int32  |

**Comment by marcosschroh** [comment]
Looks really good!! Thanks.

File: docs/fields_specification.md (hunk 1 above)
Comment (judahrand) [reply docs/fields_specification.md:28]: Done.

**Review by judahrand — COMMENTED** [review]
Requested changes made.

**Pull request merged**